core/src/test/java/org/elasticsearch/indices/template/SimpleIndexTemplateIT.java

```diff
@@ -661,4 +656,52 @@ public void testStrictAliasParsingInIndicesCreatedViaTemplates() throws Exceptio
         assertThat(response.getItems()[0].getVersion(), equalTo(1l));
     }
 
+    public void testCombineTemplates() throws Exception{
+        // clean all templates setup by the framework.
+        client().admin().indices().prepareDeleteTemplate("*").get();
+
+        // check get all templates on an empty index.
+        GetIndexTemplatesResponse response = client().admin().indices().prepareGetTemplates().get();
+        assertThat(response.getIndexTemplates(), empty());
+
+        //Now, a complete mapping with two separated templates is error
+        // base template
+        client().admin().indices().preparePutTemplate("template_1")
+            .setTemplate("*")
+            .setSettings(
+                "    {\n" +
+                    "        \"index\" : {\n" +
+                    "            \"analysis\" : {\n" +
+                    "                \"analyzer\" : {\n" +
+                    "                    \"custom_1\" : {\n" +
+                    "                        \"tokenizer\" : \"whitespace\"\n" +
+                    "                    }\n" +
+                    "                }\n" +
+                    "            }\n" +
+                    "         }\n" +
+                    "    }\n")
+            .get();
+
+        try {
+
+            // put template using custom_1 analyzer
+            client().admin().indices().preparePutTemplate("template_2")
+                    .setTemplate("test*")
+                    .setCreate(true)
+                    .setOrder(1)
+                    .addMapping("type1", XContentFactory.jsonBuilder().startObject().startObject("type1").startObject("properties")
+                        .startObject("field2").field("type", "string").field("analyzer", "custom_1").endObject()
+                        .endObject().endObject().endObject())
+                .get();
+
+            fail("mapping validation fail");
+        } catch (MapperParsingException e) {
+            assertThat(e.getMessage(), containsString("analyzer [custom_1] not found for field [field2]"));
+        }
```

Comment:
@jpountz Thanks for reviewing. One question about `expectThrows`. It is https://github.com/elastic/elasticsearch/pull/16304 , right? I should rebase first because this PR is based on the end of last year. Can I rebase first?